package.json

```diff
@@ -42,7 +42,7 @@
     "packages/*"
   ],
   "engines": {
-    "node": ">=18.14.1",
+    "node": ">=18.20.1",
```

Comment:
```suggestion
    "node": "^18.17.1 || ^20.3.0 || >=21.0.0",
    "npm": ">=9.6.5",
    "pnpm": ">=7.1.0"
```

I know this isn't the latest of 18, however the vast majority of our users are not on it (like, 98% of Node 18 users are not on it). With 17 we cover the 3 most used versions of Node 18 at this time for Astro 4.x users and make ourselves compatible with Sharp - addition of being compatible with our policy. We can raise again later on once needed / people are a bit more upgraded

Dropping 19 is part of our policy, and almost no one is on it (around 1%).

For Node 20, I know this isn't part of our policy, but in my opinion, this is safe to do because virtually no one is on those versions (it's less than 1% of users), I doubt Astro runs perfectly on those versions anyway and it makes us compatible with Sharp.

All together, these changes should affect around 9% of users, 99.99% (quite literally 20 users are on incompatible versions in CI, not sure how, might even be just a wrong detection of CI) of which are in dev.

For `npm`, older versions of `npm` already cannot install Astro correctly due to native dependencies or just modern packages, 91% of our npm users already matches this.

For `pnpm`, PNPM users are really good at keeping themselves updated, less than 1% of pnpm users don't match this.

I decided against putting a restriction for yarn just yet, but we'll add documentation to the Sharp error page about what to do for Yarn 1.

Reply:
I addressed the comment in [`d5df786` (#10689)](https://github.com/withastro/astro/pull/10689/commits/d5df7860d54f05d2e58c074f9c17c6ecbebb9a41)